plugin/dynamicforward/dynamicforward.go

```diff
@@ -0,0 +1,62 @@
+package dynamicforward
+
+import (
+	"context"
+	"fmt"
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/forward"
+	"github.com/coredns/coredns/plugin/pkg/proxy"
+	"github.com/coredns/coredns/plugin/pkg/transport"
+	"github.com/miekg/dns"
+	"log"
+	"sync"
+)
+
+// DynamicForward main struct of plugin
+type DynamicForward struct {
+	Next        plugin.Handler
+	Namespace   string
+	ServiceName string
+	forwardTo   []string
+	mu          sync.RWMutex
+	forwarder   *forward.Forward
+	options     *forward.Options
+}
+
+func (df *DynamicForward) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+	//wait forwarder
+	df.mu.RLock()
```

Comment:
There isn't guarantee the ServeDNS will be called after df.forwarder will be initialized and the mutex won't help. Need another solution.